src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs

```diff
@@ -1451,50 +1451,17 @@ void AggregateEntityResourceVaultInternal(long entityId, long resourceEntityId,
                     _ => throw new ArgumentOutOfRangeException(nameof(x), x, null),
                 };
 
-                var isDeleted = balance.IsZero();
-
                 resourceOwnersToAdd.AddOrUpdate(
-                    new ResourceOwnersLookup(x.EntityId, x.ResourceEntityId),
-                    _ =>
+                    (x.EntityId, x.ResourceEntityId),
+                    _ => new ResourceOwners
                     {
-                        if (isDeleted)
-                        {
-                            return new ResourceOwnersChange(IsDeleted: true, null);
-                        }
-
-                        return new ResourceOwnersChange(IsDeleted: false, new ResourceOwners
-                        {
-                            Id = sequences.ResourceOwnersSequence++,
-                            EntityId = x.EntityId,
-                            ResourceEntityId = x.ResourceEntityId,
-                            Balance = balance,
-                        });
+                        Id = sequences.ResourceOwnersSequence++,
+                        EntityId = x.EntityId,
+                        ResourceEntityId = x.ResourceEntityId,
+                        Balance = balance,
                     },
-                    existing =>
-                    {
-                        if (isDeleted)
-                        {
-                            existing.IsDeleted = true;
-                            existing.Entry = null;
-                            return;
-                        }
-
-                        existing.IsDeleted = false;
-
-                        if (existing.Entry != null)
-                        {
-                            existing.Entry.Balance = balance;
-                            return;
-                        }
-
-                        existing.Entry = new ResourceOwners
-                        {
-                            Id = sequences.ResourceOwnersSequence++,
-                            EntityId = x.EntityId,
-                            ResourceEntityId = x.ResourceEntityId,
-                            Balance = balance,
-                        };
-                    });
+                    existing => existing.Balance = balance
```

Comment:
I think we should keep track of the `from_state_version`, shouldn't we?

Reply:
what for? newest always wins

Reply:
Ah, right now we execute MERGE synchronously so there's no need for the `from_state_version`. However didn't we want to include `last_updated_at_state_version` in the resource owners endpoint response?

Reply:
@dhedey do we want to include that? that wasn't part of initially approved specs.

Reply:
Hm this is a good question.

Yes, I think in retrospect, thinking through the kind of insights I'd want, it would be very useful to store `last_updated_at_state_version` as an additional field (from state version is the wrong naming / abstraction).

And it would be useful to include that on the response too, if it'd be straightforward.